### ISSUE TITLE
Support for additional layers

### ISSUE THREAD

**Issue by Neelam96** [fastmachinelearning/hls4ml#176]
Does your tool provides support for PReLU, max pool and soft max.

**Comment by Duchstf** [comment]
I think PReLU, maxpooling1D and maxpooling 2D are supported for Keras. I'm not sure for softmax though. I documented the supported layers for Keras [here](https://github.com/hls-fpga-machine-learning/models/tree/master/keras). (This was updated a while ago, not sure if other people want to add anything?)

**Comment by Neelam96** [comment]
How I can add Softmax to your tool. Can u please tell me the procedure if there is one.
Thanks

**Comment by zhenbinwu** [comment]
We have a version of Softmax in [HLS](https://github.com/hls-fpga-machine-learning/hls4ml/blob/master/hls4ml/templates/vivado/nnet_utils/nnet_activation.h#L212). Might need to add the interface of Keras.

**Comment by jmduarte** [comment]
PreLU, Softmax, and MaxPooling1D/2D are all supported for Keras (and potentially PyTorch/ONNX as well).

https://github.com/abdelabd/hls4ml/blob/pyg_to_hls_rebase/hls4ml/model/hls_layers.py#L1358
https://github.com/abdelabd/hls4ml/blob/pyg_to_hls_rebase/hls4ml/model/hls_layers.py#L1371
https://github.com/abdelabd/hls4ml/blob/pyg_to_hls_rebase/hls4ml/model/hls_layers.py#L1147
https://github.com/abdelabd/hls4ml/blob/pyg_to_hls_rebase/hls4ml/model/hls_layers.py#L1178
